### PR TITLE
More robust polyfit

### PR DESCRIPTION
**Pull request by hrobotron**
Aside of polyfit, also provided more robust <code>roots()</code>.

### CODE REVIEW

**Comment by jverzani** [comment]
I'm looking at this code for `roots` and am wondering about how it could be cleaned up. There is quite a bit of work done on trimming out leading zeros. Would a call to `truncate` be sufficient for this? I'm not sure I even like this truncation, as it might be best to leave to the user to decide. (It can lead to n degree polynomials having fewer than n roots, which might be surprising.

**Comment by hrobotron** [comment]
> I'm not sure I even like this truncation, as it might be best to leave to the user to decide. (It can lead to n degree polynomials having fewer than n roots, which might be surprising.

I do not like it, either, so my vote would be to let the user decide, and perform no zero truncations. I only tried to keep the modified function to behave as close as possible to the previous version.

File: src/Polynomials.jl [hunk 1]
```diff
@@ -571,13 +564,29 @@ polyfit(xs, ys, 2)
...
+    #
+    A[:,1]=1
+    for i=1:n
+        A[:,i+1]=A[:,i] .* x   # cumulative product more precise than x.^n
+    end
+    Aqr=qrfact(A)   # returns QR object, not a matrix
+    p=Aqr\y         # least squares solution via QR
+    Poly(p, sym)
 end
 polyfit(x,y,sym::Symbol) = polyfit(x,y,length(x)-1, sym)
```
Comment (jverzani) [inline src/Polynomials.jl]: Do you want to add your 8th order example as a test to make sure there are not performance regressions going forward?

Comment (hrobotron) [inline src/Polynomials.jl]: If you don't mind merging this version without 8th order performance test, please do so -- I will try to make a test later. The .dat file has been produced through a rather complicated process, involving huge amount of consecutive FFTs and other operations. I will create something smaller and faster, and put it into another pull request later.

Comment (jverzani) [inline src/Polynomials.jl]: Okay, this looks good to me. If no one chimes in otherwise during the next day or so, I'll merge. Thanks!

**Comment by jverzani** [comment]
Thanks!

File: src/Polynomials.jl (hunk 1 above)
Comment (hrobotron) [inline src/Polynomials.jl]: I thank you.

**Pull request merged**